File: backend/services/tts/puter_voices.py
"""Puter.js TTS languages — client-side only; catalog for /api/tts/voices."""

PUTER_LANGUAGES = [
    {"id": "en-US", "name": "English (US)", "locale": "en-US"},
    {"id": "fr-FR", "name": "French", "locale": "fr-FR"},
    {"id": "de-DE", "name": "German", "locale": "de-DE"},
    {"id": "es-ES", "name": "Spanish", "locale": "es-ES"},
    {"id": "it-IT", "name": "Italian", "locale": "it-IT"},
]

DEFAULT_LANGUAGE = "en-US"


def list_assistant_voices() -> dict:
    return {
        "default_voice": DEFAULT_LANGUAGE,
        "model": "puter-js",
        "language": "multi",
        "providers": ["puter"],
        "puter_available": True,
        "groups": [
            {
                "id": "puter",
                "label": "Язык озвучки",
                "voices": [
                    {
                        "id": lang["id"],
                        "name": lang["name"],
                        "locale": lang["locale"],
                        "tier": "puter",
                        "description": "Puter TTS (браузер)",
                    }
                    for lang in PUTER_LANGUAGES
                ],
            }
        ],
    }


def tts_status_payload() -> dict:
    return {
        "enabled": True,
        "model": "puter-js",
        "provider": "puter",
        "providers": ["puter"],
        "puter_available": True,
        "language": "multi",
        "voice": DEFAULT_LANGUAGE,
        "voice_selection": True,
    }

File: frontend/lib/tts/puterVoices.ts
import type { TtsVoiceGroup } from "@/store/ttsStore";

/** Языки озвучки Puter.js — как в официальном примере. */
export const PUTER_DEFAULT_LANGUAGE = "en-US";

export const PUTER_LANGUAGES = [
  { id: "en-US", name: "English (US)", locale: "en-US" },
  { id: "fr-FR", name: "French", locale: "fr-FR" },
  { id: "de-DE", name: "German", locale: "de-DE" },
  { id: "es-ES", name: "Spanish", locale: "es-ES" },
  { id: "it-IT", name: "Italian", locale: "it-IT" },
] as const;

export type PuterLanguageId = (typeof PUTER_LANGUAGES)[number]["id"];

export const PUTER_VOICE_GROUPS: TtsVoiceGroup[] = [
  {
    id: "puter",
    label: "Язык озвучки",
    voices: PUTER_LANGUAGES.map((lang) => ({
      id: lang.id,
      name: lang.name,
      locale: lang.locale,
      tier: "puter",
      description: "Puter TTS",
    })),
  },
];

export const PUTER_PREVIEW_PHRASES: Record<PuterLanguageId, string> = {
  "en-US": "Hello! I am your Sber Business assistant.",
  "fr-FR": "Bonjour ! Je suis votre assistant Sber Business.",
  "de-DE": "Hallo! Ich bin Ihr Sber Business Assistent.",
  "es-ES": "¡Hola! Soy su asistente de Sber Business.",
  "it-IT": "Ciao! Sono il tuo assistente Sber Business.",
};

export function puterPreviewPhrase(languageId: string): string {
  return PUTER_PREVIEW_PHRASES[languageId as PuterLanguageId] ?? PUTER_PREVIEW_PHRASES["en-US"];
}
